### PR TITLE
Add IFN model for BMVC19 paper

### PR DESCRIPTION
**Pull request by BJTUJia**
Dear Kaiyang, thanks for this awesome library for the re-id community. I can't help noticing that some of your models exploit the IN layer to improve the model's generalizable ability, which is the core idea of our BMVC19 paper "Frustratingly Easy Person Re-Identification：Generalizing Person Re-ID in Practice". So I would appreciate it much if it's integrated into the library as a strong baseline, so that everyone can build on it.

### CODE REVIEW

**Comment by KaiyangZhou** [comment]
Thanks for letting me know. I was not aware of your paper. I can add your models to this repo.

I am also adding the IBN-Net code. What's the difference between your resnet_iFN and this https://github.com/XingangPan/IBN-Net/blob/master/models/imagenet/resnet_ibn_b.py?

**Comment by BJTUJia** [comment]
Many thanks! I would say the main difference to IBN_B is that we add a BN to the penultimate layer after feature encoding, which serves as feature normalization. Cheers! Thanks for your kind reply!

**Comment by KaiyangZhou** [comment]
Cool, thanks. I will read the paper. Could you update your model code to make the style consistent with, e.g., resnet.py?

It seems like your resnet_iFN and this [resnet_ibn_b](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/models/resnet_ibn_b.py) are similar?

**Comment by DukesAI** [comment]
Dear Jia,
I merged your mobilenetv_IFN to deep-person-reid by myself and trained the model according to your paper. However, I couldn't reproduce the results on "within same-domain".
I only added the following function to the bottom of your code and change the name of loss to softmax.
def mobilenetv2_ifn(num_classes, loss=None, pretrained=True, **kwargs):
    if loss is None:
        loss = {'softmax'}
    model = MobileNetV2_IFN(
        num_classes=num_classes,
        loss=loss,
        **kwargs
    )
    return model

I used SGD as optimizer which parameters are default setting of deep-person-reid.  The model was trained from scratch with initial training rate is divided by 10 after 100 epoch. The training epoch was 300. The mini-batch size was 64, and random cropping and random flipping were applied. The input of random cropping was first resized with a factor of 1.125. Label smoothing was applied to the cross entropy loss.
Could you tell me the detail of implementation if there are additional parameters or settings?
